Tests/Marketplace/search_and_install_packs.py

```diff
@@ -318,89 +335,74 @@ def install_packs(client: demisto_client,
-                    elif 'timeout awaiting response' in ex.body:
-                        raise GCPTimeOutException(ex.body)
-                    elif malformed_ids := find_malformed_pack_id(ex.body):
-                        raise MalformedPackException(malformed_ids)
-                    elif 'Item not found' in ex.body:
-                        raise GeneralItemNotFoundError(ex.body)
-                    else:
-                        raise ex
-                except Exception:
-                    logging.debug(f'An error occurred during parsing the install error: {str(ex)}')
-                    raise ex
-    try:
-        try:
-            call_install_packs_request(packs_to_install, attempts_count=3)
-
-        except MalformedPackException as e:
-            # if this is malformed pack error, remove malformed packs and retry until success
-            handle_malformed_pack_ids(e.malformed_ids, packs_to_install)
-            logging.warning(f'The request to install packs on server {host} has failed, retrying without packs '
-                            f'{e.malformed_ids}')
-            return install_packs(client, host, [pack for pack in packs_to_install if pack['id'] not in e.malformed_ids],
-                                 request_timeout)
-
-        except GCPTimeOutException as e:
-            # if this is a gcp timeout, try only once more
-            logging.warning(f'The request to install packs on server {host} has failed due to timeout awaiting response'
-                            f' headers while trying to install pack {e.pack_id}, trying again for one more time')
-            call_install_packs_request(packs_to_install)
-
-        except GeneralItemNotFoundError as e:
-            logging.warning(f'The request to install all packs on server {host} has failed due to an item not found '
-                            f'error, with the message: {e.error_msg}.\n trying again for one more time')
-            call_install_packs_request(packs_to_install)
+                if malformed_ids := find_malformed_pack_id(ex.body):
+                    handle_malformed_pack_ids(malformed_ids, packs_to_install)
+                    if not attempt:
+                        raise Exception(f"malformed packs: {malformed_ids}") from ex
 
+                    # We've more attempts, retrying without tho malformed packs.
+                    SUCCESS_FLAG = False
+                    logging.error(f"Unable to install malformed packs: {malformed_ids}, retrying without them.")
+                    packs_to_install = [pack for pack in packs_to_install if pack['id'] not in malformed_ids]
+
+                if (error_ids := get_error_ids(ex.body)) and TEST_MODELING_RULE_ERROR_CODE in error_ids:
+                    # If we got this error code, it means that the modeling rules are not valid, exiting install flow.
+                    raise Exception(f"Test modeling rules for packs {packs_to_install} are not valid") from ex
```

Comment:
This is not the right error description. 
The error code is when WLM Task ID fails. WLM Task is running validations on modeling rules and datasets.
Leave the error code in the log. We are using this error code to find the logs: https://confluence-dc.paloaltonetworks.com/display/DemistoContent/Logs+in+the+GCP#LogsintheGCP-AdvancedDebugging (Please see the video). Without the error code the developer should go to the code and find the error code and only then go to GCP. 

```suggestion
                    raise Exception(f"Failed to install: {packs_to_install}. Got [{WLM_TASK_FAILED_ERROR_CODE}] error code - Modeling rules and Dataset validations failed. Please look at GCP logs to understand why it failed.") from ex
                    
```